### ISSUE TITLE
Please bump the Golang version to a modern one

### ISSUE THREAD

**Issue by yurivict** [zmap/zgrab2#440]
[here](https://github.com/zmap/zgrab2/blob/master/go.mod#L3)

Errors:
```
# golang.org/x/sys/unix
vendor/golang.org/x/sys/unix/syscall.go:82:16: unsafe.Slice requires go1.17 or later (-lang was set to go1.16; check go.mod)
vendor/golang.org/x/sys/unix/syscall_bsd.go:247:20: unsafe.Slice requires go1.17 or later (-lang was set to go1.16; check go.mod)
vendor/golang.org/x/sys/unix/syscall_unix.go:117:7: unsafe.Slice requires go1.17 or later (-lang was set to go1.16; check go.mod)
```

**Comment by developStorm** [comment]
@phillip-stephens thoughts?

**Comment by phillip-stephens** [comment]
Yeah we should def do this. I started it [here](https://github.com/zmap/zgrab2/pull/444), but something is going wrong in the integration tests. I need to take some time and figure out how these integration tests are working to know what I broke. 😭

**Comment by mzpqnxow** [comment]
> Yeah we should def do this. I started it [here](https://github.com/zmap/zgrab2/pull/444), but something is going wrong in the integration tests. I need to take some time and figure out how these integration tests are working to know what I broke. 😭 

Do these tests pass with the current master, without changes? I vaguely recall there being some issues with CI though I don't recall of they were specifically related to validation of the output against the schemas

Something to check, before you bang your head too much against the wall 😊

**Comment by Seanstoppable** [comment]
@phillip-stephens I just realized I did a similar thing as you in #461, but integration tests pass just fine there. Looking at your own PR, it looks pretty much all the containers failed on crypto related items (you'll see errors in docker container execution). Since you didn't change any of them, perhaps a failure stemming from whatever was running the integration tests, rather than an issue with the change itself?
